File: hrms-backend/src/main/java/com/hrms/service/OfferLetterService.java

//service/OfferLetterService.java
package com.hrms.service;

import com.hrms.dto.request.OfferLetterRequest;
import com.hrms.dto.response.OfferLetterResponse;

import java.util.List;

public interface OfferLetterService {
 OfferLetterResponse create(OfferLetterRequest request);
 OfferLetterResponse update(Long id, OfferLetterRequest request);
 OfferLetterResponse getById(Long id);
 List<OfferLetterResponse> getAll();
 OfferLetterResponse updateStatus(Long id, String status, String reason);
 void delete(Long id);
 byte[] generatePdf(Long id);
}

File: hrms-frontend/src/services/offerLetterService.js
// src/services/offerLetterService.js
import api from '../api/axiosConfig';

const offerLetterService = {
  getAll: () => api.get('/offer-letters'),
  getById: (id) => api.get(`/offer-letters/${id}`),
  create: (data) => api.post('/offer-letters', data),
  update: (id, data) => api.put(`/offer-letters/${id}`, data),
  updateStatus: (id, data) => api.patch(`/offer-letters/${id}/status`, data),
  delete: (id) => api.delete(`/offer-letters/${id}`),
  downloadPdf: (id) => api.get(`/offer-letters/${id}/pdf`, {
    responseType: 'blob'
  }),
};
export default offerLetterService;